src/main/java/core/basesyntax/service/impl/CsvFruitDataReaderImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dto.FruitTransactionDto;
+import core.basesyntax.exception.CantReadFromFileException;
+import core.basesyntax.service.FruitDataReaderService;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class CsvFruitDataReaderImpl implements FruitDataReaderService {
+    @Override
+    public List<FruitTransactionDto> readData(String fileName) {
+        try {
+            List<String> strings = Files.readAllLines(Path.of(fileName));
+            FruitDataParser fruitDataParser = new FruitDataParser();
+            return fruitDataParser.parse(strings);

```

Comment:
first of all services like this parser shouldn't be created in a method like this, they should be a class fields. But in that case it should not be a part of FileReader at all, because it has 2 responsibilities - read data from file and parse in into transactions, so we should split them into 2